### ISSUE TITLE
Rule Request: Properly unwrap "implicitly unwrapped optionals"

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#3064]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues]	(https://github.com/realm/SwiftLint/issues)

### New rule request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Many of Apple's API's export "implicitly unwrapped optionals" instead of plain optionals. Usually because those APIs doesn't yet have null annotations. This is a footgun as Swift will not force you to properly unwrap it, which often cause crashes in production.

One example is [`EKCalendarItem#title`](https://developer.apple.com/documentation/eventkit/ekcalendaritem/1507305-title), which exports `var title: String! { get set }`. It's very easy to just use it directly as `event.title` instead of doing `event.title ?? ""` or guard/if.

It would be great to have a rule that forces you to explicitly unwrap "implicitly unwrapped optionals" from imported frameworks. I don't think it should trigger on "implicitly unwrapped optionals" in the same file as that's usually intentional.

Some relevant discussion: https://forums.swift.org/t/disable-objc-bridging-as-implicitly-unwrapped-optionals/21966

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Would trigger:

```swift
// event.title is EKCalendarItem#title

foo(event.title)
```

Wouldn't trigger:

```swift
foo(event.title ?? "")
```

```swift
guard let title = event.title else {
	return
}

foo(title)
````

3. Should the rule be configurable, if so what parameters should be configurable?

No

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

I would make it default as it helps catch a lot of potential crashes in production.

**Comment by iliaskarim** [comment]
A good configuration option would be to allow it to only apply to imported sources. That way developers can still use implicitly unwrapped options as they are intended to be used, that is, without any callsite unwrap.

**Comment by sindresorhus** [comment]
Please keep it open.

**Comment by sindresorhus** [comment]
Please keep it open.
